pkg/auth/authmap_gc_test.go

```diff
@@ -289,12 +292,54 @@ func Test_authMapGarbageCollector_cleanup(t *testing.T) {
 	assert.Contains(t, authMap.entries, authKey{localIdentity: 1, remoteIdentity: 2, remoteNodeID: 0, authType: policy.AuthTypeSpire})
 }
 
+func Test_authMapGarbageCollector_cleanupEndpoints(t *testing.T) {
+	ctx := context.TODO()
+
+	authMap := &fakeAuthMap{
+		entries: map[authKey]authInfo{
+			{localIdentity: 1, remoteIdentity: 2, remoteNodeID: 0, authType: policy.AuthTypeSpire}:   {expiration: time.Now().Add(5 * time.Minute)},
+			{localIdentity: 2, remoteIdentity: 1, remoteNodeID: 0, authType: policy.AuthTypeSpire}:   {expiration: time.Now().Add(5 * time.Minute)},
+			{localIdentity: 3, remoteIdentity: 1, remoteNodeID: 100, authType: policy.AuthTypeSpire}: {expiration: time.Now().Add(5 * time.Minute)},
+		},
+	}
+	gc := newAuthMapGC(logrus.New(), authMap, nil, nil,
+		&fakeEndpointRepository{
+			endpoints: []*endpoint.Endpoint{
+				{
+					SecurityIdentity: &identity.Identity{
+						ID: 2,
+					},
+				},
+				{
+					SecurityIdentity: &identity.Identity{
+						ID: 3,
+					},
+				},
+			},
+		},
+	)
+
+	assert.Len(t, authMap.entries, 3)
+
+	gc.ciliumIdentitiesDiscovered = map[identity.NumericIdentity]struct{}{
+		1: {},
+		2: {},
+		3: {},
+	}
+	gc.ciliumIdentitiesSynced = true
+
+	err := gc.cleanupEndpoints(ctx)
+	assert.NoError(t, err)
+	assert.Len(t, authMap.entries, 1, "GC runs should delete all entries where the secrity ID no longer is in the endpoint map")
+	assert.Contains(t, authMap.entries, authKey{localIdentity: 3, remoteIdentity: 1, remoteNodeID: 100, authType: policy.AuthTypeSpire})
```

Comment:
Maybe also a test of the noop case - that if you run `cleanupEndpoints` and all the relevant identities are still present in the endpoint map, nothing is deleted?